STMicroelectronics/STM32L4_L4+/app/startup/gnu/STM32L475xx_FLASH.ld

```diff
@@ -47,9 +47,15 @@ MEMORY
 {
 RAM (xrw)      : ORIGIN = 0x20000000, LENGTH = 96K
 RAM2 (xrw)      : ORIGIN = 0x10000000, LENGTH = 32K
-FLASH (rx)      : ORIGIN = 0x8000000, LENGTH = 1024K
+FLASH (rx)      : ORIGIN = 0x8000000, LENGTH = 1020K
+REGISTRYINFO (rx)      : ORIGIN = 0x80FD800, LENGTH = 2K
+REGISTRY (rx)      : ORIGIN = 0x80FF000, LENGTH = 4K
```

Comment:
I would not split the flash as you did here, only create a new section after `.rodata`.

Reply:
Something like this after `.rodata`:
```
  /* Registry data goes into FLASH */
  .regdata :
  {
    . = ALIGN(8);
    *(.regdata)         /* .regdata sections (list of key value pair.) */
    *(.regdata*)        /* .regdata* sections (list of key value pai.) */
    . = ALIGN(8);
  } >FLASH
```

Reply:
With that, you can create variables directly in this section using `__attribute__((section (".regdata")))`

Reply:
Same for the other processor.

Reply:
Marcos and I discussed offline and agreed to keep it as is for now to prevent 1) registry getting erased when new code is flashed 2) when using linker script to save a section in FLASH, the initialized value is 0. Will not be spending time on this iteration of the registry to solve this issue but we may come back and revisit.